diff -ruN a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2026-06-16 01:51:46
+++ b/src/CMakeLists.txt	2026-06-16 01:51:46
@@ -147,6 +147,25 @@
 lyx_find_info_files(LyXCMakeFiles ${TOP_SCRIPT_PATH}/*)
 lyx_find_info_files(LyXUiFiles ${TOP_SRC_DIR}/lib/ui/*)
 
+set(lyx_mathd_sources ${lyx_sources})
+list(REMOVE_ITEM lyx_mathd_sources
+	${TOP_SRC_DIR}/src/main.cpp
+	${TOP_SRC_DIR}/src/HunspellChecker.cpp)
+
+add_library(lyxcore_for_mathd OBJECT
+	${lyx_mathd_sources}
+	${lyx_headers}
+)
+
+add_dependencies(lyxcore_for_mathd lyx_version)
+
+set_target_properties(lyxcore_for_mathd PROPERTIES
+  QT_NO_UNICODE_DEFINES TRUE
+	FOLDER "applications/LyX"
+	INTERPROCEDURAL_OPTIMIZATION ${LYX_IPO_SUPPORTED})
+
+qt_use_modules(lyxcore_for_mathd Core Gui ${QtCore5CompatModule})
+
 add_executable(${_lyx}
 	${WIN32_CONSOLE}
 	${LYX_BUILD_BUNDLE}
diff -ruN a/src/convert/CMakeLists.txt b/src/convert/CMakeLists.txt
--- a/src/convert/CMakeLists.txt	1970-01-01 08:00:00
+++ b/src/convert/CMakeLists.txt	2026-06-16 01:51:46
@@ -0,0 +1,25 @@
+# This file is part of LyX, the document processor.
+# Licence details can be found in the file COPYING.
+
+project(${_convert})
+
+include_directories(BEFORE
+	"${TOP_SRC_DIR}/src/convert")
+
+add_executable(${_convert}
+	"${TOP_SRC_DIR}/src/convert/lyxconvert.cpp")
+
+target_link_libraries(${_convert}
+	${QT_QTCORE_LIBRARY}
+	${QT_QTGUI_LIBRARY}
+	${QT_QTWIDGETS_LIBRARY}
+	${ICONV_LIBRARY}
+	${ZLIB_LIBRARY})
+
+qt_use_modules(${_convert} Core Gui Widgets)
+
+if(APPLE)
+	target_link_libraries(${_convert} "-framework AppKit")
+endif()
+
+install(TARGETS ${_convert} DESTINATION ${LYX_UTILITIES_INSTALL_PATH})
